### ISSUE TITLE
Dockerfiles don't work on Skosmos 2.18

### ISSUE THREAD

**Issue by matthew-morris-cessda** [NatLibFi/Skosmos#1678]
## URL address of the page where you encountered the problem

N/A

## Description of the problem

Building the Docker image for Skosmos 2.18 fails with the following message

```text
...
#17 [13/16] RUN php composer.phar install --no-dev --no-autoloader
#17 0.621 No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
#17 0.622 Loading composer repositories with package information
#17 1.831 Updating dependencies
#17 1.879 Your requirements could not be resolved to an installable set of packages.
#17 1.879 
#17 1.879   Problem 1
#17 1.879     - Root composer.json requires sweetrdf/easyrdf 1.13.* -> satisfiable by sweetrdf/easyrdf[1.13.0, 1.13.1].
#17 1.879     - sweetrdf/easyrdf[1.13.0, ..., 1.13.1] require php ^8.0 -> your php version (7.4.3) does not satisfy that requirement.
#17 1.879 
#17 1.879 Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.
#17 ERROR: executor failed running [/bin/sh -c php composer.phar install --no-dev --no-autoloader]: exit code: 2
------
 > [13/16] RUN php composer.phar install --no-dev --no-autoloader:
0.621 No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
0.622 Loading composer repositories with package information
1.831 Updating dependencies
1.879 Your requirements could not be resolved to an installable set of packages.
1.879 
1.879   Problem 1
1.879     - Root composer.json requires sweetrdf/easyrdf 1.13.* -> satisfiable by sweetrdf/easyrdf[1.13.0, 1.13.1].
1.879     - sweetrdf/easyrdf[1.13.0, ..., 1.13.1] require php ^8.0 -> your php version (7.4.3) does not satisfy that requirement.
1.879 
1.879 Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.
------
ERROR: failed to solve: executor failed running [/bin/sh -c php composer.phar install --no-dev --no-autoloader]: exit code: 2
```

This appears to be due to using Ubuntu 20.04 as the base image which provides PHP 7.4, which is no longer supported.

## The browser you used when the problem appeared

N/A

**Comment by osma** [comment]
Thank you for the issue report. I think you're right, PHP 7.4 is the problem here and the base image needs to be changed or updated.

**Comment by osma** [comment]
We already have PRs #1272 and #1294 that probably would fix the problem by upgrading to a newer base image. They have unfortunately been left open without merging.

**Comment by osma** [comment]
The problem has been fixed in PR #1680 and the fix has been released in [Skosmos 2.8.1](https://github.com/NatLibFi/Skosmos/releases/tag/v2.18.1). Closing the issue.
